src/back_end.rs

```diff
@@ -10,78 +12,79 @@ const UV_COMPONENTS: usize = 2;
 // Needs to be improved on gfx-rs side.
 // For some reason, using ``*_COMPONENT` triggers some macros errors.
 
-gfx_vertex!( PositionFormat {
-    pos@ pos: [f32; 2],
+gfx_vertex_struct!( PositionFormat {
+    pos: [f32; 2] = "pos",
 });
 
-gfx_vertex!( ColorFormat {
-    color@ color: [f32; 4],
+gfx_vertex_struct!( ColorFormat {
+    color: [f32; 4] = "color",
 });
 
-gfx_vertex!( TexCoordsFormat {
-    uv@ uv: [f32; 2],
+gfx_vertex_struct!( TexCoordsFormat {
+    uv: [f32; 2] = "uv",
 });
 
-gfx_parameters!( Params {
-    color@ color: [f32; 4],
+gfx_pipeline!( pipe_colored {
+    pos: gfx::VertexBuffer<PositionFormat> = (),
+    color: gfx::Global<[f32; 4]> = "color",
+    out_color: gfx::RenderTarget<gfx::format::Rgba8> = "o_Color",
 });
 
-gfx_parameters!( ParamsUV {
-    color@ color: [f32; 4],
-    s_texture@ texture: gfx::shade::TextureParam<R>,
+gfx_pipeline!( pipe_textured {
+    pos: gfx::VertexBuffer<PositionFormat> = (),
+    uv: gfx::VertexBuffer<TexCoordsFormat> = (),
+    color: gfx::Global<[f32; 4]> = "color",
+    texture: gfx::TextureSampler<[f32; 4]> = "s_texture",
+    out_color: gfx::RenderTarget<gfx::format::Rgba8> = "o_Color",
+    blend_target: gfx::BlendTarget<gfx::format::Rgba8> =
```

Comment:
well there is your problem, totally unexpected by GFX - using two RTV component with the same name, and binding the same object there